### PR TITLE
chore(swagger): Init swagger

### PR DESCRIPTION
**Pull request by lauthieb**

### CODE REVIEW

File: src/main/java/fr/foodlimit/api/config/SwaggerConfig.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package fr.foodlimit.api.config;
+
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
```
Comment (lauthieb) [inline src/main/java/fr/foodlimit/api/config/SwaggerConfig.java]: MINOR Remove this unused import 'org.springframework.boot.autoconfigure.EnableAutoConfiguration'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/fr/foodlimit/api/config/SwaggerConfig.java [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.ComponentScan;
```
Comment (lauthieb) [inline src/main/java/fr/foodlimit/api/config/SwaggerConfig.java]: MINOR Remove this unused import 'org.springframework.context.annotation.ComponentScan'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/fr/foodlimit/api/config/SwaggerConfig.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+import org.springframework.context.annotation.Configuration;
+import org.springframework.web.servlet.config.annotation.EnableWebMvc;
```
Comment (lauthieb) [inline src/main/java/fr/foodlimit/api/config/SwaggerConfig.java]: MINOR Remove this unused import 'org.springframework.web.servlet.config.annotation.EnableWebMvc'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/fr/foodlimit/api/config/SwaggerConfig.java [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+import org.springframework.web.servlet.config.annotation.ResourceHandlerRegistry;
```
Comment (lauthieb) [inline src/main/java/fr/foodlimit/api/config/SwaggerConfig.java]: MINOR Remove this unused import 'org.springframework.web.servlet.config.annotation.ResourceHandlerRegistry'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/fr/foodlimit/api/config/SwaggerConfig.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  @Bean
+  public Docket api() {
+    return new Docket(DocumentationType.SWAGGER_2)
+      .select()
+      .apis(RequestHandlerSelectors.any())
+      .paths(PathSelectors.any())
+      .build();
+  }
+
+  private ApiInfo apiInfo() {
```
Comment (lauthieb) [inline src/main/java/fr/foodlimit/api/config/SwaggerConfig.java]: MAJOR Remove this unused private "apiInfo" method. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AUnusedPrivateMethod)

**Comment by lauthieb** [comment]
SonarQube analysis reported 7 issues
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AppConfig.java#L11](https://github.com/food-limit/food-limit-api/blob/3025534e894cc000d4cc9658eac6d3a1b244d55d/src/main/java/fr/foodlimit/api/config/AppConfig.java#L11): Remove this unused "secret" private field. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [AppConfig.java#L13](https://github.com/food-limit/food-limit-api/blob/3025534e894cc000d4cc9658eac6d3a1b244d55d/src/main/java/fr/foodlimit/api/config/AppConfig.java#L13): Remove this unused "tokenValidityInSeconds" private field. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1068)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by lauthieb** [comment]
SonarQube analysis reported 7 issues
* ![MAJOR][MAJOR] 3 major
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [AppConfig.java#L11](https://github.com/food-limit/food-limit-api/blob/73d9a0517ec78532b3103fa08e75b0ba8995bfab/src/main/java/fr/foodlimit/api/config/AppConfig.java#L11): Remove this unused "secret" private field. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1068)
1. ![MAJOR][MAJOR] [AppConfig.java#L13](https://github.com/food-limit/food-limit-api/blob/73d9a0517ec78532b3103fa08e75b0ba8995bfab/src/main/java/fr/foodlimit/api/config/AppConfig.java#L13): Remove this unused "tokenValidityInSeconds" private field. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1068)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by landschootl — APPROVED** [review]

**Pull request merged**